### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by sv-Tjipto**
This pull request adds StyleGAN2 on implementation on ADNI Brain dataset with all the required functionality without the dataset that can be found on the link on the README.md. This repository includes dataset.py, modules.py, utils.py, predict.py, train.py and config.py.

### CODE REVIEW

**Comment by Uto0op** [comment]
Recognition problem: StyleGAN on ADNI ---- Hard Difficulty

1. Algorithm solves the problem: (3.5 Marks)
Not very clear generated images, it's not reasonable to resize to 64 * 64 as lots of information would be lost -0.5;
TSNE embedding on the training data without brief interpretation/discussion -1

2. Algorithm demonstrated and functions: (3 Marks)

4. Design: 
Very modularized design (1 Mark)

5. Commenting:
Sufficient docstrings and comments (1 Mark)

6. Algorithm above Normal Difficulty: (5 Marks)

7. Algorithm is Hard difficulty: (5 Marks)

**Suggestions**:
- Resizing from 256*256 down to 64*64 is not appropriate, causing the generated images being too small to identify with information lost;
- Ideally, according to the reference paper from the task sheet, the embedding should be plotted on the style space. In your `train.py`, you plotted the embedding directly on the training data by using `get_dataloader(FIXED_IMAGE_SIZE, args.batch_size, args.data_root, shuffle=False)` with `args.data_roo` set to the train dir on Rangpur (-0.5). Also, a brief interpretation/discussion is required (-0.5);
- The losses shown are fine, but the gap suggests it could be further fine-tuned;

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Good solution to problem. </td><td></td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard. StyleGAN (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. Some messages are abstract</td><td>-1</td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. TSNE  interpretation missing. No loss.</td><td>-1</td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Pull Request has problems: Late submission.</td><td>-2</td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-5</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by hanemma7moud** [comment]
Timothy Tjipto

**Comment by shakes76** [comment]
Late submission -10 (2 days). Email me with evidence to restore if you have an extension.
